# Superstore Analysis
## Explorative Data Analysis
* **Goal:** TBA

In [2]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
# Safe data loading with try/except
file_path = '../data/superstore.csv'

try:
    df = pd.read_csv(file_path, encoding='latin1')
    print(f'Data loaded successfully')
except FileNotFoundError:
    print(f'File not found - check {file_path}')
except Exception as e:
    print(f'Unexpected error occurred - check {e}')

Data loaded successfully


In [4]:
# First inspection of Dataset with df.info()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [5]:
# View the head of the dataset
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [6]:
# Inspect the Statistic Metrics of Dataset
df.describe()

,Row ID,Postal Code,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000,9994.000000
mean,4997.500000,55190.379428,229.858001,3.789574,0.156203,28.656896
std,2885.163629,32063.693350,623.245101,2.225110,0.206452,234.260108
min,1.000000,1040.000000,0.444000,1.000000,0.000000,-6599.978000
25%,2499.250000,23223.000000,17.280000,2.000000,0.000000,1.728750
50%,4997.500000,56430.500000,54.490000,3.000000,0.200000,8.666500
75%,7495.750000,90008.000000,209.940000,5.000000,0.200000,29.364000
max,9994.000000,99301.000000,22638.480000,14.000000,0.800000,8399.976000


In [7]:
# Check columns of the dataset
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [8]:
# Checking for missing values
df.isna().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [9]:
# Check for duplicates
df.duplicated().sum()

np.int64(0)

**Observation:** The dataset contains no missing values and no duplicate records.
However, the descriptive statistics reveal unusually low profit values, with a minimum profit of -$6,599.98. This will be investigated further to determine whether these losses are caused by discounts, specific products, or other business factors.

In [10]:
# Convert 'Order Date' and 'Ship Date' into pandas datetime
df['Order Date'] = pd.to_datetime(df['Order Date'], format='%m/%d/%Y')
df['Ship Date'] = pd.to_datetime(df['Ship Date'], format='%m/%d/%Y')

# Check the type for both
print(df['Order Date'].dtype)
print(df['Ship Date'].dtype)


datetime64[ns]
datetime64[ns]


In [11]:
df[['Order Date', 'Ship Date']].head()

,Order Date,Ship Date
0,2016-11-08,2016-11-11
1,2016-11-08,2016-11-11
2,2016-06-12,2016-06-16
3,2015-10-11,2015-10-18
4,2015-10-11,2015-10-18


In [12]:
# Convert 'Postal Code' into string
df['Postal Code'] = df['Postal Code'].astype(str).str.zfill(5)

# Check for dtype of 'Postal Code'
print(df['Postal Code'].dtype)

object


In [13]:
df['Postal Code'].head()

0    42420
1    42420
2    90036
3    33311
4    33311
Name: Postal Code, dtype: object

**Data Type Adjustments:**
- Converted Order Date and Ship Date to datetime format.
- Converted Postal Code to string format to preserve leading zeros.

In [14]:
# First analysis of most comun category
print(df.groupby('Category')['Sales'].agg(['count', 'sum', 'mean']).sort_values(by='sum', ascending=False).round(2))

                 count        sum    mean
Category                                 
Technology        1847  836154.03  452.71
Furniture         2121  741999.80  349.83
Office Supplies   6026  719047.03  119.32


**Analysis:**  
The analysis shows that **Office Supplies** has the highest number of transactions, while **Technology** generates the highest total revenue despite having fewer transactions. This indicates that Technology products have a significantly higher average sales value per order.

**Recommendation:**  
Since Office Supplies accounts for the highest transaction volume, it may offer opportunities for revenue growth. However, further analysis of pricing, customer demand, and price elasticity should be conducted before considering any price adjustments.

In [15]:
# Check for the most comun sub category
print(df.groupby('Sub-Category')['Sales'].agg(['count', 'sum', 'mean']).sort_values(by='sum', ascending=False).round(2))

              count        sum     mean
Sub-Category                           
Phones          889  330007.05   371.21
Chairs          617  328449.10   532.33
Storage         846  223843.61   264.59
Tables          319  206965.53   648.79
Binders        1523  203412.73   133.56
Machines        115  189238.63  1645.55
Accessories     775  167380.32   215.97
Copiers          68  149528.03  2198.94
Bookcases       228  114880.00   503.86
Appliances      466  107532.16   230.76
Furnishings     957   91705.16    95.83
Paper          1370   78479.21    57.28
Supplies        190   46673.54   245.65
Art             796   27118.79    34.07
Envelopes       254   16476.40    64.87
Labels          364   12486.31    34.30
Fasteners       217    3024.28    13.94
